File: Backend/AccountingSystem/src/AccountingSystem.Domain/Entities/Companyreportconfig.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AccountingSystem.Domain.Entities;


public partial class Companyreportconfig
{

    public int Id { get; set; }

    public int Companyid { get; set; }

    public int Reporttypeid { get; set; }

    public int Frequencyid { get; set; }

    public short? Dayofmonth { get; set; }

    public bool? Isactive { get; set; }

    public DateTime? Createdat { get; set; }

    public DateTime? Updatedat { get; set; }

    public virtual Company Company { get; set; } = null!;
    public virtual Frequency Frequency { get; set; } = null!;
    public virtual ICollection<Reportinstance> Reportinstances { get; set; } = new List<Reportinstance>();
    public virtual Reporttype Reporttype { get; set; } = null!;
}

File: Fronted/Angular-app/src/models/company-report-config.ts
export interface CompanyReportConfigDto {
  id: number;
  companyId: number;
  reportTypeId: number;
  frequencyId: number;
  dayOfMonth?: number;
  isActive?: boolean;
  createdAt: Date;
  updatedAt: Date;
  year: number;

  companyName?: string;
  reportTypeName?: string;
  reportTypeShortCode?: string;
  frequencyName?: string;
}

export interface CreateCompanyReportConfigDto {
  companyId: number;
  reportTypeId: number;
  frequencyId: number;
  dayOfMonth?: number;
  year: number;
}

export interface UpdateCompanyReportConfigDto extends CreateCompanyReportConfigDto {
  id: number;
}
